Library/DasharoSystemFeaturesUiLib/DasharoSystemFeatures.c

```diff
@@ -230,48 +231,51 @@ DasharoSystemFeaturesUiLibConstructor (
   if (!PRIVATE_DATA(HybridCpuArchitecture))
     PRIVATE_DATA(CoreMaxCount) = PRIVATE_DATA(BigCoreMaxCount);
 
-#define LOAD_VAR(var, field) do {                                                   \
-    BufferSize = sizeof (mDasharoSystemFeaturesPrivate.DasharoFeaturesData.field);  \
-    Status = gRT->GetVariable (                                                     \
-        (var),                                                                      \
-        &gDasharoSystemFeaturesGuid,                                                \
-        NULL,                                                                       \
-        &BufferSize,                                                                \
-        &mDasharoSystemFeaturesPrivate.DasharoFeaturesData.field                    \
-        );                                                                          \
-    ASSERT_EFI_ERROR (Status);                                                      \
+#define LOAD_VAR(var, field, var_type) do {                           \

```

Comment:
Thanks, was not aware if that is possible...

Reply:
@krystian-hebel still not exactly sure how I can use _Generic in the macro. From what I understood it maps the returned type of the function taking arguments of various types. I have no idea how to map that to struct fields.

```C
#define GetDasharoDefault(var) _Generic((var), \
                    BOOLEAN: DasharoGetVariableDefault(var).Boolean,  \
                      UINT8: DasharoGetVariableDefault(var).Uint8,    \
    DASHARO_WATCHDOG_CONFIG: DasharoGetVariableDefault(var).Watchdog, \
       DASHARO_IOMMU_CONFIG: DasharoGetVariableDefault(var).Iommu,    \
     DASHARO_BATTERY_CONFIG: DasharoGetVariableDefault(var).Battery   \
              )(var) 
```

I came up with this, but it doesn't make sense, because there must be a function name only as an expression in the association list.

Reply:
I think it should be something similar to this at the bottom of `LOAD_VAR()` macro:
```c
PRIVATE_DATA(field) = _Generic(PRIVATE_DATA(field), \
                    BOOLEAN: DasharoGetVariableDefault(var).Boolean,  \
                      UINT8: DasharoGetVariableDefault(var).Uint8,    \
    DASHARO_WATCHDOG_CONFIG: DasharoGetVariableDefault(var).Watchdog, \
       DASHARO_IOMMU_CONFIG: DasharoGetVariableDefault(var).Iommu,    \
     DASHARO_BATTERY_CONFIG: DasharoGetVariableDefault(var).Battery   \
    )
```
`(X)` in example is used because `_Generic` picks a function which is then invoked.

Reply:
Dang it... The compiler says UINT8 and BOOLEAN are compatible types and there is multiple match...

Reply:
a778203cf3625447caf4c42d60f8d2dede1f1ad2

left comment about the boolean in the commit message